shared_objects/static_function.c

```diff
@@ -0,0 +1,52 @@
+#include "include/static_function.h"
+#include "include/find_sentries.h"
+#include <assert.h>
+#include <stdio.h>
+
+static int static_function()
+{
+	assert(base_and_bounds_eq(cheri_pcc_get(), &static_function));
+	return 0;
+}
+
+static double another_static_function()
+{
+	assert(base_and_bounds_eq(cheri_pcc_get(), &another_static_function));
+	return 1.0;
+}
+
+void non_static_fun()
+{
+	assert(base_and_bounds_eq(cheri_pcc_get(), &non_static_fun));
+	assert(base_and_bounds_eq(cheri_pcc_get(), &static_function));
+	printf("Called static function with result: %d\n", static_function());
+}
+
+void non_static_fun2()
+{
+	assert(base_and_bounds_eq(cheri_pcc_get(), &non_static_fun2));
+	assert(base_and_bounds_eq(cheri_pcc_get(), &another_static_function));
+	printf("Called another function with result: %f\n", another_static_function());
+}
+
+void check_static_and_non_static()
+{
+	assert(base_and_bounds_eq(cheri_pcc_get(), &non_static_fun));
+	assert(base_and_bounds_eq(cheri_pcc_get(), &static_function));
+	assert(base_and_bounds_eq(cheri_pcc_get(), &non_static_fun2));
+	assert(base_and_bounds_eq(cheri_pcc_get(), &another_static_function));
+	assert(base_and_bounds_eq(cheri_pcc_get(), &check_static_and_non_static));
+	printf("Static and non static checks...\n\n");
```

Comment:
Let's only have `print`s in tests if they're important (e.g. to help debug likely problems). I don't know which are and aren't important so I will leave the selection to you :)

Reply:
They probably are not important at all. We can remove them.

Reply:
Excellent!

Reply:
Fix in 91f2d41.